src/com/aca/narekhakobyan/homework1/items/concepts/RecordableGadgetWithoutMemory.java

```diff
@@ -0,0 +1,37 @@
+package com.aca.narekhakobyan.homework1.items.concepts;
+
+import java.util.Objects;
+import java.util.Scanner;
+
+public abstract class RecordableGadgetWithoutMemory extends Gadget {
+    protected final Integer recDuration;
```

Comment:
think about this property again,  and maybe you can put validation for the name property? In any case, I don't like that in the lowest levels of your abstraction you don't have concrete models, I will add requirements which will show the needing of that models, for instance, "writing colors of pen" or smth like that.

Reply:
@garikKalash